File: sources/Violence b1.26/de/violence/module/modules/MOVEMENT/Hypixel.java
package de.violence.module.modules.MOVEMENT;

import de.violence.module.Module;
import de.violence.module.ui.Category;
import net.minecraft.network.play.client.C03PacketPlayer;

public class Hypixel extends Module {
   public Hypixel() {
      super("Hypixel", Category.MOVEMENT);
   }

   public void onUpdate() {
      this.mc.thePlayer.motionY = 0.0D;
      if(this.mc.thePlayer.ticksExisted % 3 == 0) {
         this.mc.thePlayer.sendQueue.addToSendQueue(new C03PacketPlayer.C04PacketPlayerPosition(this.mc.thePlayer.posX, this.mc.thePlayer.posY - 1.0E-10D, this.mc.thePlayer.posZ, true));
      }

      this.mc.thePlayer.setPosition(this.mc.thePlayer.posX, this.mc.thePlayer.posY + 1.0E-10D, this.mc.thePlayer.posZ);
      this.mc.thePlayer.onGround = true;
      this.mc.gameSettings.keyBindJump.pressed = false;
      super.onUpdate();
   }
}

File: sources/Kevin Reborn/kevin/module/modules/movement/speeds/other/Hypixel.kt
package kevin.module.modules.movement.speeds.other

import kevin.event.UpdateEvent
import kevin.module.BooleanValue
import kevin.module.ListValue
import kevin.module.modules.movement.speeds.SpeedMode
import kevin.utils.MovementUtils
import kevin.utils.RandomUtils
import net.minecraft.client.settings.GameSettings

object Hypixel : SpeedMode("Hypixel") {
    private val mode = ListValue("${valuePrefix}Mode", arrayOf("LowHop", "FastDown"), "FastDown")
    private val airStrafe = BooleanValue("${valuePrefix}AirStrafe", true)
    private val modeValue: Boolean
        get() = mode equal "LowHop"
    var airTicks = 0
    override fun onUpdate(event: UpdateEvent) {
        val thePlayer = mc.thePlayer ?: return
        if (!MovementUtils.isMoving) {
            mc.gameSettings.keyBindJump.pressed = GameSettings.isKeyDown(mc.gameSettings.keyBindJump)
            airTicks = 5
            return
        }
        if (thePlayer.onGround) {
            mc.gameSettings.keyBindJump.pressed = true
            MovementUtils.strafe()
            airTicks = 0
            if (modeValue) {
                thePlayer.motionY -= RandomUtils.random.nextDouble() * 0.01
            }
        } else if (!modeValue) {
            if (++airTicks == 4) {
                thePlayer.motionY = -0.10200565429997936
                if (airStrafe.get()) {
                    MovementUtils.strafe()
                }
            }
        }
    }

    override fun onDisable() {
        mc.gameSettings.keyBindJump.pressed = GameSettings.isKeyDown(mc.gameSettings.keyBindJump)
    }
}